redis/tests/test_basic.rs

```diff
@@ -915,6 +926,72 @@ mod basic {
         assert_eq!(msg.get_payload(), Ok("bar".to_string()));
     }
 
+    #[test]
+    fn test_pubsub_send_ping() {
+        use std::sync::{Arc, Barrier};
+        let ctx = TestContext::new();
+        let mut con = ctx.connection();
+
+        // Connection for subscriber api
+        let mut pubsub_con = ctx.connection();
+
+        // Barrier is used to make test thread wait to publish
+        // until after the pubsub thread has subscribed.
+        let publish_barrier = Arc::new(Barrier::new(2));
```

Comment:
nit
It's not wrong, but `Barrier` seems like an untraditional solution, why not `mspc`?

Reply:
In general, barriers are pretty traditiontal - if anything, out of Rust they're much more common than channels. :)

But more relevant to this specific test - notice that the barriers are used for 2-directional blocking. The barriers block either thread until the other thread reaches the barrier. If I wanted to implement this with channels, I'd have complex code that needs to account for the order in which the threads reached the point of waiting. So Barrier is the abstraction that better fits the problem domain for this test.